src/components/accessRequests/accessRequestsList/accessRequestsList.tsx

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+// for the German Human Genome-Phenome Archive (GHGA)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useCallback, useEffect, useState } from "react";
+import { AccessRequest } from "../../../models/submissionsAndRequests";
+import { Col, Row, Table } from "react-bootstrap";
+import AccessRequestModal from "./accessRequestModal";
+import { User } from "../../../services/auth";
+import SortButton, { TableFields } from "../../../utils/sortButton";
+import { transposeTableForHTML } from "../../../utils/utils";
+
+//
+interface AccessRequestListProps {
+  requests: AccessRequest[];
+  setRequests: any;
+  user: User;
+}
+
+const AccessRequestsList = (props: AccessRequestListProps) => {
+  const buildInnerTable = useCallback(() => {
+    return [
+      {
+        header: "ID",
+        data: props.requests.map((x) => x.id),
+        cssClasses: "d-none",
+      },
+      {
+        header: "Dataset",
+        data: props.requests.map((x) => x.dataset_id),
+      },
+      {
+        header: "User",
+        data: props.requests.map((x) => x.full_user_name),
+      },
+      {
+        header: "Starts",
+        data: props.requests.map((x) => x.access_starts.split("T")[0]),
+      },
+      {
+        header: "Ends",
+        data: props.requests.map((x) => x.access_ends.split("T")[0]),
+      },
+      {
+        header: "Requested",
+        data: props.requests.map((x) => x.request_created.split("T")[0]),
+      },
+      {
+        header: "Status",
+        data: props.requests.map((x) => x.status),
+      },
+    ];
+  }, [props.requests]);
+
+  const [innerTable, setInnerTable] = useState<TableFields[]>(
+    buildInnerTable()
+  );
+
+  useEffect(() => {
+    let table: TableFields[] = buildInnerTable();
+    setInnerTable(table);
+    setSortedData(transposeTableForHTML(table.map((x) => x.data)));
+  }, [buildInnerTable, props.requests]);
+
+  const [sortDefinition, setSortDefinition] = useState<{
+    key: number;
+    order: number;
+  }>({
+    key: 0,
+    order: 0,
+  });
+
+  const [sortedData, setSortedData] = useState<any>(
+    transposeTableForHTML(innerTable.map((x) => x.data))
+  );
+
+  let tableDefinition = {
+    sortDefinition,
+    setSortDefinition,
+    sortedData,
+    setSortedData,
+    table: innerTable,
+  };
+
+  function onUpdate() {
+    if (props.requests) {
+      props.setRequests([...props.requests]);
```

Comment:
Probably it's better to not use a handler `setRequests` but `onChangeRequest` that passes a single (modified) request back. The wrapper function then replaces that request in the array of all requests.

Reply:
Done